File: 卡码/代码随想录/1/05-栈与队列/Go/02-删除字符串所有相邻重复项/LeetCode_1047.go
//1047. 删除字符串中的所有相邻重复项
package main

func removeDuplicates(s string) string {
	stack:=make([]rune,0)
	for _,ch:=range s {
		if len(stack)==0 || stack[len(stack)-1]!=ch {
			stack=append(stack,ch)
		}else{
			stack=stack[0:len(stack)-1]
		}
	}
	var str string=""
	for _,ch:=range stack {
		str+=string(ch)
	}
	return str
}

File: 卡码/代码随想录/1/05-栈与队列/Java/src/_02_删除字符串中所有相邻重复项/LeetCode_1047.java
//1047. 删除字符串中的所有相邻重复项
package _02_删除字符串中所有相邻重复项;

import java.util.Stack;

public class LeetCode_1047 {

}
class Solution {
  public String removeDuplicates(String s) {
  	Stack<Character> stack = new Stack<>();
  	for(int i=0;i<s.length();i++) {
  		if(stack.empty()||s.charAt(i)!=stack.peek()) stack.push(s.charAt(i));
  		else {
  			if(stack.peek()==s.charAt(i)) stack.pop();
  		}
  	}
  	StringBuilder builder = new StringBuilder();
  	while(!stack.empty()) {
  		builder.append(stack.pop());
  	}
  	return new String(builder.reverse());
  }
}